pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/CustomSerDeUtils.java

```diff
@@ -243,9 +243,14 @@ public TDigest deserialize(ByteBuffer byteBuffer) {
 
     @Override
     public byte[] serialize(Sketch value) {
-      // NOTE: Compact the sketch in unsorted, on-heap fashion for performance concern.
-      //       See https://datasketches.apache.org/docs/Theta/ThetaSize.html for more details.
-      return value.compact(false, null).toByteArray();
+      // The serializer should respect existing ordering to enable "early stop"
+      // optimisations on unions.
+      boolean shouldCompact = !value.isCompact();
+      boolean shouldOrder = value.isOrdered();
```

Comment:
Shouldn't we order if the value is not ordered?
`boolean shouldOrder = value.isOrdered() -> !value.isOrdered()`?

Current code just passes `shouldOrder` to compact(). In that case, we can just pass `compact(value.isOrdered(), null)`?

Reply:
I was simply commenting that
```
boolean shouldOrder = value.isOrdered();
if (shouldCompact) {
  return value.compact(shouldOrder, null).toByteArray();
}
```
is the same as
```
if (shouldCompact) {
  return value.compact(value.isOrdered(), null).toByteArray();
}
```
Since `shouldCompact` was not being used in this function. If `value.isOrdered()` is provided by the end user to indicate that we should sort, we can keep it as is.

Reply:
Yes, the ordering is set by the end user.